apps/studio/src/features/editing-experience/components/RootStateDrawer.tsx

```diff
@@ -122,7 +154,56 @@ export default function RootStateDrawer() {
     ],
   )
 
-  const pageLayout: string = savedPageState.layout
+  const handleConversionToIndexPage = useCallback(() => {
+    const newPageState = {
+      ...savedPageState,
+      content: [],
+      childpages: {
+        layout: "boxes",
+        summary: false,
+        thumbnail: false,
+      },
+    }
+    // NOTE: This layout needs to be outside, otherwise it is treated as a
+    // string type rather than the array of layout consts
+    newPageState.layout = "index"
+    newPageState.content = []
+    setPreviewPageState(newPageState)
+    setIsPreviewingIndexPage(true)
+  }, [savedPageState, setPreviewPageState])
+
+  const handleCancelConversionToIndexPage = useCallback(() => {
+    setIsPreviewingIndexPage(false)
+    setPreviewPageState(savedPageState)
+  }, [savedPageState, setPreviewPageState])
+
+  const handleSaveConversionToIndexPage = useCallback(() => {
+    savePage(
+      {
+        pageId,
+        siteId,
+        content: JSON.stringify(previewPageState),
+      },
+      {
+        onSuccess: () => {
+          setIsPreviewingIndexPage(false)
+          setSavedPageState(previewPageState)
+          onConfirmConvertIndexPageModalClose()
+          setDrawerState({ state: "root" })
+        },
+      },
+    )
+  }, [
+    onConfirmConvertIndexPageModalClose,
+    pageId,
+    previewPageState,
+    savePage,
+    setDrawerState,
+    setSavedPageState,
+    siteId,
+  ])
+
+  const pageLayout: string = previewPageState.layout
```

Comment:
[nit] can we omit the `string` typing so it infers the correct type

Reply:
Updated!